installer/packer/scripts/harbor/configure_harbor.sh

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/bash
+# Copyright 2017 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -x -euf -o pipefail
+
+deploy=$(ovfenv -k harbor.deploy)
+
+if [ ${deploy,,} != "true" ]; then
+  echo "Not configuring Harbor" && exit 0
+fi
+
+cert_dir=/data/harbor/cert
+flag=/etc/vmware/harbor/cert_gen_type
+cfg=/data/harbor/harbor.cfg
+
+ca_download_dir=/data/harbor/ca_download
+mkdir -p {${cert_dir},${ca_download_dir}}
+
+cert=${cert_dir}/server.crt
+key=${cert_dir}/server.key
+csr=${cert_dir}/server.csr
+ca_cert=${cert_dir}/ca.crt
+ca_key=${cert_dir}/ca.key
+ext=${cert_dir}/extfile.cnf
+
+rm -rf $ca_download_dir/*
+
+#Configure attr in harbor.cfg
+function configureHarborCfg {
+  cfg_key=$1
+  cfg_value=$2
+
+  basedir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+  if [ -n "$cfg_key" ]
+  then
+    cfg_value=$(echo "$cfg_value" | sed -r -e 's%[\/&%]%\\&%g')
+    sed -i -r "s%#?$cfg_key\s*=\s*.*%$cfg_key = $cfg_value%" $cfg
+  fi
+}
+
+function format {
+  file=$1
+  head=$(sed -rn 's/(-+[A-Za-z ]*-+)([^-]*)(-+[A-Za-z ]*-+)/\1/p' $file)
+  body=$(sed -rn 's/(-+[A-Za-z ]*-+)([^-]*)(-+[A-Za-z ]*-+)/\2/p' $file)
+  tail=$(sed -rn 's/(-+[A-Za-z ]*-+)([^-]*)(-+[A-Za-z ]*-+)/\3/p' $file)
+  echo $head > $file
+  echo $body | sed  's/\s\+/\n/g' >> $file
+  echo $tail >> $file
+}
+
+function genCert {
+  if [ ! -e $ca_cert ] || [ ! -e $ca_key ]
+  then
+    openssl req -newkey rsa:4096 -nodes -sha256 -keyout $ca_key \
+      -x509 -days 365 -out $ca_cert -subj \
+      "/C=US/ST=California/L=Palo Alto/O=VMware, Inc./OU=Containers on vSphere/CN=Self-signed by VMware, Inc."
+  fi
+  openssl req -newkey rsa:4096 -nodes -sha256 -keyout $key \
+    -out $csr -subj \
+    "/C=US/ST=California/L=Palo Alto/O=VMware/OU=Containers on vSphere/CN=$hostname"
+
+  echo "Add subjectAltName = IP: $ip_address to certificate"
+  echo subjectAltName = IP:$ip_address > $ext
+  openssl x509 -req -days 365 -in $csr -CA $ca_cert -CAkey $ca_key -CAcreateserial -extfile $ext -out $cert
+
+  echo "self-signed" > $flag
+  echo "Copy CA certificate to $ca_download_dir"
+  cp $ca_cert $ca_download_dir/
+}
+
+function secure {
+  ssl_cert=$(ovfenv -k harbor.ssl_cert)
+  ssl_cert_key=$(ovfenv -k harbor.ssl_cert_key)
+  if [ -n "$ssl_cert" ] && [ -n "$ssl_cert_key" ]; then
+    echo "ssl_cert and ssl_cert_key are both set, using customized certificate"
```

Comment:
That's a great idea, I'll revise the script in the next iteration to check for that.